client/state/jetpack-connect/utils.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Internal dependencies
+ */
+import { JETPACK_CONNECT_TTL } from './constants';
+
+/***
+ * Whether a Jetpack Connect store timestamp is stale.
+ * @param   {Number} timestamp  Item to check.
+ * @returns {Boolean}           True if the timestamp is stale, false otherwise.
+ */
+export function isStale( timestamp ) {
+	const now = new Date().getTime();
+	if ( ! timestamp ) {
```

Comment:
There shouldn't be any effect on the users who are in the middle of the auth process - if they don't have a timestamp, their authorization data will not be considered stale, so they'll be able to continue the process. And we're not validating the authorization state with the schema at this time, so this will not result in unexpected session resets.